### ISSUE TITLE
Calling `isOptional()` on a `coerce.bigint` throws an error

### ISSUE THREAD

**Issue by marcus13371337** [colinhacks/zod#1911]
Try the following code:
```ts
const bigintSchema = z.coerce.bigint();
const isOptional = bigintSchema.isOptional();
```

That code should be able to run IMO, but now it throws an error, `TypeError: Cannot convert undefined to a BigInt`

**Comment by marcus13371337** [comment]
Kind of related to the following:
https://github.com/colinhacks/zod/discussions/1897
https://github.com/colinhacks/zod/discussions/1856

I can't really see the benefit of why `Zod` shouldn't catch these errors and handle them in the same way as you use to with most of the other errors that can occur. What are the drawbacks of having `Zod` a bit smarter and having this as the default implementation of `coerce.bigint`?

```ts
        try {
            return BigInt( value )
        } catch ( error ) {
            ctx.addIssue( {
                code: 'invalid_type',
                expected: 'unknown',
                received: value,
                message: `Can't be parsed to BigInt`,
            } )
        }
```

**Comment by JacobWeisenburger** [comment]
@marcus13371337 
Was my answer satisfactory? or would you like to leave this issue open as an enhancement request?

**Comment by marcus13371337** [comment]
Yes, I was able to get around the problem by creating a custom bigint-schema. But I don't see why `zod`, shouldn't be able to reply of a schema is optional without crashing (which happens to be a coerce bigint), to me that's clearly a bug

**Comment by RobinTail** [comment]
same about `isNullable`

```ts
const f = z.coerce.bigint();
f.isNullable();
```

**Comment by RobinTail** [comment]
looks like this issue will be fixed in Zod 4 — it does not throw in this case.

**Comment by colinhacks** [comment]
Closing this out. This no longer reproduces in current Zod: z.coerce.bigint().isOptional() still exists and returns false instead of throwing.
